src/vs/workbench/contrib/notebook/browser/controller/editActions.ts

```diff
@@ -646,44 +647,44 @@ registerAction2(class CommentSelectedCellsAction extends NotebookMultiCellAction
 	async runWithContext(accessor: ServicesAccessor, context: INotebookCommandContext): Promise<void> {
 		const languageConfigurationService = accessor.get(ILanguageConfigurationService);
 
-		const selectedCellEditors: ICodeEditor[] = [];
-		context.selectedCells.forEach(cell => {
-			const findContext = { notebookEditor: context.notebookEditor, cell };
-			const foundEditor = findTargetCellEditor(findContext, cell);
-			if (foundEditor) {
-				selectedCellEditors.push(foundEditor);
-			}
-		});
-
-
-		selectedCellEditors.forEach(editor => {
-			if (!editor.hasModel()) {
+		context.selectedCells.forEach(async cellViewModel => {
+			const cellTextModel = cellViewModel.model;
+			if (!cellTextModel) {
 				return;
 			}
+			const textBuffer = cellTextModel.textBuffer;
 
-			const model = editor.getModel();
-			const commands: ICommand[] = [];
-			const modelOptions = model.getOptions();
-			const commentsOptions = editor.getOption(EditorOption.comments);
-
-			const selection = editor.getSelection();
+			let textModel: ITextModel | TextModel | undefined = cellTextModel.textModel;
+			if (!textModel) {
+				textModel = await cellViewModel.resolveTextModel();
+			}
```

Comment:
maybe we could just do `textModel = await cell.resolveTextModel();` since `resolve*` can early return when it already exists.